File: SCT2/trunk/test-plugins/org.yakindu.sct.generator.java.test/test-gen/org/yakindu/sct/generator/java/test/BooleanExpressionsTest.java
/**
 * Copyright (c) 2013 committers of YAKINDU and others.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 *
 * Contributors:
 *     committers of YAKINDU - initial API and implementation
 */
package org.yakindu.sct.generator.java.test;

import org.junit.After;
import org.junit.Before;
import org.junit.Test;
import static org.junit.Assert.*;
import org.yakindu.scr.booleanexpressions.BooleanExpressionsStatemachine;
import org.yakindu.scr.booleanexpressions.BooleanExpressionsStatemachine.State;
/**
 *  Unit TestCase for BooleanExpressions
 */
@SuppressWarnings("all")
public class BooleanExpressionsTest {

	private BooleanExpressionsStatemachine statemachine;

	@Before
	public void setUp() {
		statemachine = new BooleanExpressionsStatemachine();
		statemachine.init();
	}

	@After
	public void tearDown() {
		statemachine = null;
	}

	@Test
	public void testbooleanExpressions() {
		statemachine.enter();
		assertTrue(statemachine.isStateActive(State.main_region_StateA));
		assertTrue(statemachine.getMyBool1() == true);
		assertTrue(statemachine.getMyBool2() == false);
		statemachine.raiseE1();
		statemachine.runCycle();
		assertTrue(statemachine.isStateActive(State.main_region_StateB));
		assertTrue(statemachine.getAnd() == false);
		assertTrue(statemachine.getOr() == true);
		assertTrue(statemachine.getNot() == false);
		assertTrue(statemachine.getEqual() == false);
		assertTrue(statemachine.getNotequal() == true);
	}
}

File: test-plugins/org.yakindu.sct.generator.c.test/gtests/BooleanExpressions/BooleanExpressionsTest.cc
/**
* Copyright (c) 2017 committers of YAKINDU and others.
* All rights reserved. This program and the accompanying materials
* are made available under the terms of the Eclipse Public License v1.0
* which accompanies this distribution, and is available at
* http://www.eclipse.org/legal/epl-v10.html
*
* Contributors:
*     committers of YAKINDU - initial API and implementation
*/
#include "gtest/gtest.h"
#include "BooleanExpressions.h"

static BooleanExpressions statechart;

class StatemachineTest : public ::testing::Test{
	protected:
	virtual void SetUp() {
		booleanExpressions_init(&statechart);
	}
};


TEST_F(StatemachineTest, booleanExpressions) {					
	booleanExpressions_enter(&statechart);
	EXPECT_TRUE(booleanExpressions_isStateActive(&statechart, BooleanExpressions_main_region_StateA));
	EXPECT_TRUE(booleanExpressionsIface_get_myBool1(&statechart)== true);
	EXPECT_TRUE(booleanExpressionsIface_get_myBool2(&statechart)== false);
	booleanExpressionsIface_raise_e1(&statechart);
	booleanExpressions_runCycle(&statechart);
	EXPECT_TRUE(booleanExpressions_isStateActive(&statechart, BooleanExpressions_main_region_StateB));
	EXPECT_TRUE(booleanExpressionsIface_get_and(&statechart)== false);
	EXPECT_TRUE(booleanExpressionsIface_get_or(&statechart)== true);
	EXPECT_TRUE(booleanExpressionsIface_get_not(&statechart)== false);
	EXPECT_TRUE(booleanExpressionsIface_get_equal(&statechart)== false);
	EXPECT_TRUE(booleanExpressionsIface_get_notequal(&statechart)== true);
}


